File: BulkyBookCoreMVC/wwwroot/js/company.js
var dataTable;
$(document).ready(function () {

    loadDataTable();
});

function loadDataTable() {
    dataTable = $('#tblData').DataTable({
        "ajax": {
            "url": "/Admin/Company/GetAllCompany"
        },
        "columns": [
            { "data": "name", "width": "10%" },
            { "data": "streetAddress", "width": "10%" },
            { "data": "city", "width": "10%" },
            { "data": "state", "width": "10%" },
            { "data": "postalCode", "width": "10%" },
            { "data": "phoneNumber", "width": "10%" },
            {
                "data": "id",
                "render": function (data) {
                    return `
                     <div class="w-60 btn-group" role="group">
                        <a href=/Admin/Company/Edit?id=${data} 
                        class="btn btn-primary mx-2"> <i class="bi bi-pencil-square"></i>Edit</a>

                        <a href=/Admin/Company/Delete?id=${data}
                        class="btn btn-primary mx-2"> <i class="bi bi-trash-fill"></i>Delete </a>
                     </div>
                    `


                },

                "width":"15%"


             }
            
            

            
        ]  

    });
}

File: BulkyBookCoreMVC/Models/Company.cs

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BulkyBookCoreMVC.Models
{
    public class Company
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [Required]
        public string? Name {  get; set; }

        public string? StreetAddress { get; set; }

        public string? City { get; set; }

        public string? State { get; set; }
        public string? PostalCode { get; set; }

        public string? PhoneNumber {  get; set; }
    }
}
